lib/failure.sh

```diff
@@ -115,31 +115,53 @@ fail_iojs_unsupported() {
 }
 
 fail_multiple_lockfiles() {
+  local build_dir="${1:-}"
   local has_modern_lockfile=false
-  if [ -f "${1:-}/yarn.lock" ] || [ -f "${1:-}/package-lock.json" ]; then
-    has_modern_lockfile=true
-  fi
 
-  if [ -f "${1:-}/yarn.lock" ] && [ -f "${1:-}/package-lock.json" ]; then
-    mcount "failures.two-lock-files"
-    meta_set "failure" "two-lock-files"
+  declare -A lockfiles=(
+    ["npm"]="package-lock.json"
+    ["pnpm"]="pnpm-lock.yaml"
+    ["Yarn"]="yarn.lock"
+  )
+
+  local detected_package_managers=()
+  for package_manager in "${!lockfiles[@]}"; do
+    lockfile="${lockfiles["$package_manager"]}"
+    if [ -f "$build_dir/$lockfile" ]; then
+      has_modern_lockfile=true
+      detected_package_managers+=("$package_manager")
+    fi
+  done
+
+  if (( "${#detected_package_managers[*]}" > 1 )); then
+    readarray -td '' package_managers_sorted < <(printf '%s\0' "${detected_package_managers[@]}" | sort -z --ignore-case)
+    mcount "failures.multiple-lock-files"
+    meta_set "failure" "multiple-lock-files"
     header "Build failed"
-    warn "Two different lockfiles found: package-lock.json and yarn.lock
+    warn "Multiple lockfiles found
 
-       Both npm and yarn have created lockfiles for this application,
+       Multiple package managers ($(IFS=','; printf '%s' "${package_managers_sorted[*]}")) have created lockfiles for this application,
        but only one can be used to install dependencies. Installing
        dependencies using the wrong package manager can result in missing
        packages or subtle bugs in production.
 
        - To use npm to install your application's dependencies please delete
-         the yarn.lock file.
+         the pnpm and/or Yarn file.
 
-         $ git rm yarn.lock
+         $ git rm ${lockfiles["pnpm"]}
+         $ git rm ${lockfiles["Yarn"]}
 
-       - To use yarn to install your application's dependencies please delete
-         the package-lock.json file.
+       - To use Yarn to install your application's dependencies please delete
+         the npm and/or pnpm lockfiles.
 
-         $ git rm package-lock.json
+         $ git rm ${lockfiles["npm"]}
+         $ git rm ${lockfiles["pnpm"]}
+
+       - To use pnpm to install your application's dependencies please delete
```

Comment:
Not blocking:

Adding this third bullet makes sense given the previous message, but is this total message overly exhaustive now? It's little long and hard to grok. Is this better?

```
Multiple package managers ($(IFS=','; printf '%s' "${package_managers_sorted[*]}")) have created lockfiles for this application, but only one can be used to install dependencies. Installing dependencies using the wrong package manager can result in missing packages or subtle bugs in production.

Only one of the following package manager lockfiles are supported at a time: ${lockfiles["npm"], ${lockfiles["yarn"]}, ${lockfiles["pnpm"]}. Please delete the lockfile(s) that should not be in use.
```

Reply:
Yeah, this message is a bit much 😅 

Updated it based on your suggestion.